pkg/watcher/certificate.go

```diff
@@ -0,0 +1,263 @@
+package watcher
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/go-logr/logr"
+	"github.com/kyma-project/lifecycle-manager/api/v1alpha1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	metav1 "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	apimachinerymetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// private key will only be generated if one does not already exist in the target `spec.secretName`.
+	privateKeyRotationPolicy = "Never"
+
+	DomainAnnotation  = "skr-domain"
+	CertificateSuffix = "-watcher-certificate"
+
+	caCertKey        = "ca.crt"
+	tlsCertKey       = "tls.crt"
+	tlsPrivateKeyKey = "tls.key"
+)
+
+var LabelSet = k8slabels.Set{ //nolint:gochecknoglobals
+	v1alpha1.PurposeLabel: v1alpha1.CertManager,
+	v1alpha1.ManagedBy:    v1alpha1.OperatorName,
+}
+
+type SubjectAltName struct {
+	DNSNames       []string
+	IPAddresses    []string
+	URIs           []string
+	EmailAddresses []string
+}
+
+type Certificate struct {
```

Comment:
This is imo not really well named since its not representing a certificate resource, its representing a struct containing pointers that are necessary to edit certificates. Maybe call it CertificateSync?

Reply:
I think `CertificateSync` is more misleading than `Certificate`, since there is not sync happening.

The `Certificate` is a struct which has all necessary information to create a single cert-manager certificate for one single Kyma instance. Meaning, this is not meant to be for multiple KymaCRs or to sync some sort of state.

I tried to come up with a different name, but for me the most sufficient is `Certificate` - What do you think? or do you may have a different proposal than `CertificateSync`?

Reply:
Understood the proposition now I think. Suggestion: A Certificate is a Certificate, the thing that creates/deletes a single cert-manager Certificate is a `CertificateManager` with a Method called `Create` and a method called `Delete` which deal with the certificate and only get passed the `context`.

You also don't need to pass `config *SkrChartManagerConfig` into Create if its already in the `CertificateManager`

WDYT?